### PR TITLE
Update making-your-own-starboard.md

### PR DESCRIPTION
**Pull request by zophiale**
v12 update

### CODE REVIEW

**Review by cfanoulis — APPROVED** [review]
LGTM!

File: coding-guides/making-your-own-starboard.md [hunk 1]
```diff
@@ -135,15 +135,15 @@ module.exports = class {
     if (message.author.id === user.id) return message.channel.send(`${user}, you cannot star your own messages.`);
     if (message.author.bot) return message.channel.send(`${user}, you cannot star bot messages.`);
     const { starboardChannel } = this.client.settings.get(message.guild.id);
-    const starChannel = message.guild.channels.find(channel => channel.name === starboardChannel)
+    const starChannel = message.guild.channels.cache.find(channel => channel.name === starboardChannel)
     if (!starChannel) return message.channel.send(`It appears that you do not have a \`${starboardChannel}\` channel.`); 
     const fetchedMessages = await starChannel.fetchMessages({ limit: 100 });
     const stars = fetchedMessages.find(m => m.embeds[0].footer.text.startsWith('⭐') && m.embeds[0].footer.text.endsWith(message.id));
```
Comment (eslachance) [inline coding-guides/making-your-own-starboard.md:141]: Doesn't v12 use something like channel.messages.fetch now?

File: coding-guides/making-your-own-starboard.md [hunk 2]
```diff
@@ -156,7 +156,7 @@ module.exports = class {
     if (!stars) {
       const image = message.attachments.size > 0 ? await this.extension(reaction, message.attachments.array()[0].url) : '';
       if (image === '' && message.cleanContent.length < 1) return message.channel.send(`${user}, you cannot star an empty message.`);
-      const embed = new RichEmbed()
+      const embed = new MessageEmbed()
         .setColor(15844367)
         .setDescription(message.cleanContent)
         .setAuthor(message.author.tag, message.author.displayAvatarURL)
```
Comment (eslachance) [inline coding-guides/making-your-own-starboard.md:162]: Check this and others, I think v12 now has most avatar/icon things as methods now, like displayAvatarUrl()

**Review by eslachance — CHANGES_REQUESTED** [review]
Please make sure to go through each line and make sure methods and properties haven't changed. Other than displayAvatarUrl() and the fetch, there might be other things that have changed that were missed here.

File: coding-guides/making-your-own-starboard.md [hunk 3]
```diff
@@ -218,15 +218,15 @@ module.exports = class {
...
     const stars = fetchedMessages.find(m => m.embeds[0].footer.text.startsWith('⭐') && m.embeds[0].footer.text.endsWith(reaction.message.id));
     if (stars) {
       const star = /^\⭐\s([0-9]{1,3})\s\|\s([0-9]{17,20})/.exec(stars.embeds[0].footer.text);
       const foundStar = stars.embeds[0];
       const image = message.attachments.size > 0 ? await this.extension(reaction, message.attachments.array()[0].url) : '';
-      const embed = new RichEmbed()
+      const embed = new MessageEmbed()
         .setColor(foundStar.color)
         .setDescription(foundStar.description)
         .setAuthor(message.author.tag, message.author.displayAvatarURL)
```
Comment (ghost) [inline coding-guides/making-your-own-starboard.md:232]: The avatar props have been moved to a function, yeah?

File: coding-guides/making-your-own-starboard.md (hunk 1 above)
Comment (zophiale) [reply coding-guides/making-your-own-starboard.md:141]: nope, it uses message.guild.channels.cache.

File: coding-guides/making-your-own-starboard.md (hunk 2 above)
Comment (zophiale) [reply coding-guides/making-your-own-starboard.md:162]: ok!

File: coding-guides/making-your-own-starboard.md [hunk 4]
```diff
@@ -218,15 +218,15 @@ module.exports = class {
     if (message.author.id === user.id) return;
     if (reaction.emoji.name !== '⭐') return;
     const { starboardChannel } = this.client.settings.get(message.guild.id);
-    const starChannel = message.guild.channels.find(channel => channel.name == starboardChannel)
+    const starChannel = message.guild.channels.cache.find(channel => channel.name == starboardChannel)
     if (!starChannel) return message.channel.send(`It appears that you do not have a \`${starboardChannel}\` channel.`); 
     const fetchedMessages = await starChannel.fetchMessages({ limit: 100 });
```
Comment (ghost) [inline coding-guides/making-your-own-starboard.md:223]: > 
> 
> Doesn't v12 use something like channel.messages.fetch now?

File: coding-guides/making-your-own-starboard.md (hunk 3 above)
Comment (zophiale) [reply coding-guides/making-your-own-starboard.md:232]: I shall make a new PR to resolve this issue...

File: coding-guides/making-your-own-starboard.md (hunk 4 above)
Comment (zophiale) [reply coding-guides/making-your-own-starboard.md:223]: nope

File: coding-guides/making-your-own-starboard.md [hunk 5]
```diff
@@ -191,7 +191,7 @@ if (stars) {
   const star = /^\⭐\s([0-9]{1,3})\s\|\s([0-9]{17,20})/.exec(stars.embeds[0].footer.text);
   const foundStar = stars.embeds[0];
   const image = message.attachments.size > 0 ? await this.extension(reaction, message.attachments.array()[0].url) : '';
-  const embed = new RichEmbed()
+  const embed = new MessageEmbed()
     .setColor(foundStar.color)
     .setDescription(foundStar.description)
     .setAuthor(message.author.tag, message.author.displayAvatarURL)
```
Comment (ghost) [inline coding-guides/making-your-own-starboard.md:197]: > The avatar props have been moved to a function, yeah?

File: coding-guides/making-your-own-starboard.md (hunk 4 above)
Comment (YorkAARGH) [reply coding-guides/making-your-own-starboard.md:223]: Looking at the official documentation for _version 12_, they do not have a `fetchMessages` method anymore and Zinc is correct, it's been changed to `channel.messages.fetch`

File: coding-guides/making-your-own-starboard.md (hunk 1 above)
Comment (YorkAARGH) [reply coding-guides/making-your-own-starboard.md:141]: Evie was asking about the removed `fetchMessages()` method that's no longer present according to the v12 documentation.

File: coding-guides/making-your-own-starboard.md (hunk 5 above)
Comment (YorkAARGH) [reply coding-guides/making-your-own-starboard.md:197]: They have, but this PR hasn't been updated yet, nor a new one opened like he said.

**Comment by ghost** [comment]
This file has been updated completely in #102. This PR may be closed.

**Pull request closed without merging**